### PR TITLE
Added support for Django 4.1

### PR DESCRIPTION
**Pull request by mlodic**

### CODE REVIEW

**Comment by mlodic** [comment]
I added this. I am trying to have this explicitly support new Django versions so we can use it in modern Django projects.

**Comment by cdriehuys** [comment]
Thanks for the PR! I'd be willing to cut down the supported versions in order to get Django 4 compatibility. From [Django's docs](https://www.djangoproject.com/download/#supported-versions), it looks like the 3.2 LTS and any Django 4 version is all that's supported right now.

**Comment by cdriehuys** [comment]
Great, thanks. I think we could also decrease the number of test environments needed. Based on [Django's supported Python versions](https://docs.djangoproject.com/en/dev/faq/install/#what-python-version-can-i-use-with-django), I think we could get away with:

| Django Version | Python Version(s) |
| ----------------|------------------ |
| 3.2 | 3.6, 3.10 |
| 4.x | 3.8, 3.11 |

File: setup.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def get_description():
     packages=find_packages(),
     # Dependencies
     install_requires=[
-        "Django >= 2.2, < 3.2",
+        "Django >= 2.2, <= 4.2",
```
Comment (cdriehuys) [inline setup.py:41]:
```suggestion
        "Django >= 3.2, <= 4.2",
```

File: docs/CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,8 @@ In Development
 
 Features
   * :issue:`96`: Ability to customize email subjects.
+  * :issue:`97`: Added support for Django 4. We explicitly test against Django
+    2.2, 3.2, and 4.0, but all versions after 2.2 should work.
```
Comment (cdriehuys) [inline docs/CHANGELOG.rst:10]:
```suggestion
  * :issue:`97`: Added support for Django 4. We explicitly test against Django
    3.2, 4.0, and 4.1, but all versions after 3.2 should work.
```

File: setup.py [hunk 3]
```diff
@@ -38,7 +38,7 @@ def get_description():
...
         "django-email-utils >= 1.0, < 2.0",
         # DRF 3.10 is the first to support our minimum Django version of 2.2.
         "djangorestframework >= 3.10",
```
Comment (cdriehuys) [inline setup.py:44]: DRF 3.14 is the first version to support Django 3.2, 4.0, and 4.1.

```suggestion
        "djangorestframework >= 3.14",
```

**Review by cdriehuys — CHANGES_REQUESTED** [review]

**Review by cdriehuys — APPROVED** [review]

**Comment by cdriehuys** [comment]
Looks like `black` is failing. This should really be separated into another build since we don't need to check the style for every Python version. I can take a look at doing this later if you don't mind waiting a bit.

**Comment by cdriehuys** [comment]
I went ahead and merged this with the tests failing. I'll clean those up and then cut a release when I get a chance. Thanks again for the PR.

**Comment by mlodic** [comment]
thank you!

**Pull request merged**